### DISCUSSION TITLE
EUID Wallet Architecture and Reference Framework

### DISCUSSION THREAD

**Discussion by christianheimann** [e-id-admin/governance-sounding-board#42] [General]
Die EU hat eine erste Version des Architecture and Reference Frameworks publiziert:
https://digital-strategy.ec.europa.eu/en/library/european-digital-identity-wallet-architecture-and-reference-framework

Es sind noch nicht alle Fragen politisch abschliessend geklärt, aber es zeigt die Richtung, in die Europa geht.
Was fällt euch auf? Was ist genial? Wo sollten wir lernen? Was sollten wir vermeiden?

**Comment by pascalgottret** [comment]
Sorry für die späte Antwort. Irgendwie habe ich keine Nachricht für diesen Thread per Mail bekommen.

Habe das ARF in Ruhe durchgelesen und finde den Ansatz sehr gut. Folgendes möchte ich anbringen:

- es kommt in gewissen Bereichen selbstverständlich rüber, dass Verifikatoren (z.B. im eCommerce-Bereich) ohne Weiteres PID abfragen dürfen. Es steht in den Rolen von den Relying Parties, dass PID abgefragt werden kann "on their own decision". Genau das ist wegen Überidentifikation sehr gefährlich. 
- Das ARF verlangt, dass Verifikatoren registriert sein müssen. Ich sehe das als unrealistisch und kontraproduktiv. Ich denke, das wurde auch von diversen Parteien im Rahmen des BGeID kritisiert.
- Welche zugrundeliegende Technologie für globale Identifikatoren verwendet wird, wird nicht verdeutlich. Entsprechend unklar, ob Decentralized Identifiers nach W3C Standard verwendet werden sollen.
- Spannend ist die Aussage "Cryptographic key management to prevent export of cryptoggraphic secrets." ist das eine reine Sicherheitsmassnahme oder will man verhindern, dass PID auf mehreren Geräten abgelegt ist und damit das "Binding Problem" etwas zu umgehen?
- Auch interessant den Verweis auf mDL (Mobile Driving License) von ISO/IEC. Damit stellt man die Interoperabilität sicher, allerdings nur für den einen Usecase. Wie sinnvoll das ist, kann ich nicht beurteilen.

Was sind andere Meinungen?

Am 2. März findet übrigens über das EUDI Wallet ein kostenloses Webinar statt.

**Reply by pascalgottret** [reply]
Hier noch ein Paar Ergänzung, da ja das Framework auf den eIDAS Regularien basiert:

- der Non-Repudiation Ansatz ist absolut begrüssenswert. Also alle Anfragen von Verifiern müssen von ihnen selbst signiert werden, so dass er zu einem späteren Zeitpunkt nicht abstreiten kann, dass gewisse Daten abgefragt wurden.
- Bin mir nicht sicher, ob das zwingend bestimmt wurde: ein Credential muss zwingend mit mehreren (z.B. 10-50) Identifiers verknüpft sein, so dass beim Präsentieren des Credentials immer ein anderer Identifier verwendet wird. Eine "Correlation" wird dadurch nochmal erschwert. Das finde ich ein absolutes Must-Have!

**Reply by romanzoun** [reply]
> * Das ARF verlangt, dass Verifikatoren registriert sein müssen. Ich sehe das als unrealistisch und kontraproduktiv. Ich denke, das wurde auch von diversen Parteien im Rahmen des BGeID kritisiert.

Faktisch machen sie das, weil sie DSGVO nicht durchkriegen. So zwingen sie das auf. Jeder verifier muss dann DSGVO konform sein, um in die liste zu kommen und gleichzeitig die meisten müssen 🙈. Ich finde aber das sehr merkwürdig und wird nicht aufgehen, da nicht jedes Mitgliedstaat gleichermassen demokratisch ist^^

**Reply by romanzoun** [reply]
> * Auch interessant den Verweis auf mDL (Mobile Driving License) von ISO/IEC. Damit stellt man die Interoperabilität sicher, allerdings nur für den einen Usecase. Wie sinnvoll das ist, kann ich nicht beurteilen.

Also das ist für PID daten....und nun ist es Auslegungssache. PID ist ja auch mein Diplom von der Uni, da steht mein Name und Geburtsdatum drauf usw...Also ich glaube das ding fällt den Machern über die Füsse, denn am Ende hast du alles in mDL und nur das google login, mit der fake email-adresse in type 2. Ich finde es falsch hier einen Unterschied zu machen. 

mDL an sich wird verwendet, da es auch dann offline funktioniert ohne Internet.

**Comment by pascalgottret** [comment]
Es gibt übrigens ebenfalls eine Github-Seite für das Framework:
https://github.com/eu-digital-identity-wallet/architecture-and-reference-framework

Hier kann man mitdiskutieren.
